ds3-sdk/src/main/java/com/spectralogic/ds3client/metadata/MetaDataAccessImpl.java

```diff
@@ -0,0 +1,156 @@
+                        metadata.put("x-amz-meta-ds3-owner", owner.getName());
+                        break;
+
+                    case "user":
+                        break;
+
+                    case "unix":
+                        break;
+
+                    case "posix":
+                        final PosixFileAttributes attr1 = Files.readAttributes(file, PosixFileAttributes.class);
+                        // metadata.put("x-amz-meta-ds3-owner", attr1.owner().getName());
+                        metadata.put("x-amz-meta-ds3-groupName", attr1.group().getName());
+                        metadata.put("x-amz-meta-ds3-permisions", getModes(PosixFilePermissions.toString(attr1.permissions())));
+                        break;
+                }
+            }
+        } catch (final IOException e) {
+            e.printStackTrace();
+        }
+        return metadata;
+    }
+
+    // get the octal number for the permission
+    private String getModes(String permissions) {
+        final String permString = new String(permissions);
+        permissions = permissions.replaceAll("r", "4");
+        permissions = permissions.replaceAll("w", "2");
+        permissions = permissions.replaceAll("x", "1");
+        permissions = permissions.replaceAll("-", "0");
+        final String ownerPerm = String.valueOf(Integer.parseInt(String.valueOf(permissions.charAt(0))) + Integer.parseInt(String.valueOf(permissions.charAt(1))) + Integer.parseInt(String.valueOf(permissions.charAt(2))));
+        final String groupPerm = String.valueOf(Integer.parseInt(String.valueOf(permissions.charAt(3))) + Integer.parseInt(String.valueOf(permissions.charAt(4))) + Integer.parseInt(String.valueOf(permissions.charAt(5))));
+        final String otherPerm = String.valueOf(Integer.parseInt(String.valueOf(permissions.charAt(6))) + Integer.parseInt(String.valueOf(permissions.charAt(7))) + Integer.parseInt(String.valueOf(permissions.charAt(8))));
+        final String totalPerm = ownerPerm + groupPerm + otherPerm;
+        return totalPerm + "(" + permString + ")";
+    }
+
+    //if os is windows then posix will not be called and we need to find permission in different manner
+    private void saveWindowsfilePermissions(final Path file, final Map<String, String> metadata) {
+        try {
+            final AclFileAttributeView view = Files.getFileAttributeView(file, AclFileAttributeView.class);
+            final List<AclEntry> acl = view.getAcl();
+            Set<AclEntryPermission> aclEntryPermissions;
+            String userType = "";
+            StringBuilder permission = null;
+
+            for (final AclEntry ac : acl) {
+                permission = new StringBuilder();
+                int index = 1;
+                aclEntryPermissions = ac.permissions();
+
+                for (final AclEntryPermission aclEntryPermission : aclEntryPermissions) {
+                    if (index == aclEntryPermissions.size())
```

Comment:
Please add explicit curly braces to conform with our java coding standard.  Here is a link to the specific section on the topic: [Curly Braces](https://github.com/SpectraLogic/spectralogic.github.com/wiki/Java-Style-Guide#conditional-statements)